### PR TITLE
[118] Quick Fixes and Validation Issues with Server Config File

### PR DESCRIPTION
**Pull request by keithchong**

### CODE REVIEW

File: dev/com.ibm.ws.st.ui/src/com/ibm/ws/st/ui/internal/Activator.java [hunk 1]
```diff
@@ -222,31 +223,54 @@ public void run() {
         });
 
         com.ibm.ws.st.core.internal.Activator.setFeatureConflictHandler(new FeatureConflictHandler() {
+
+            WebSphereServerInfo wsServerInfo = null;
```
Comment (eharris369) [inline dev/com.ibm.ws.st.ui/src/com/ibm/ws/st/ui/internal/Activator.java]: These fields are not safe.  Even if reset to null when invokeDialog is completed it is still not safe if multiple threads are calling this method at the same time.

**Comment by eharris369** [comment]
Comments:

- In this case there is a runtime (potentially a 'ghost' runtime) that has two user directories each with one server.  One of the user directories and servers corresponds to the classical liberty setup (this is the target/build) and the other one does not (this is the src).  In order to get the src user directory and server to behave well they probably need to extend the existing UserDirectory and WebSphereServerInfo classes so that the extended server configuration files (bootstrap.properties, server.env, jvm.options) can be found (they don't have to be in the same directory as server.xml), includes can be resolved (again they don't have to be in the same directory as server.xml), and variables in the pom.xml/build.gradle can be handled better (maybe the CustomServerVariablesHandler won't be needed anymore either).
- Could the new runtimes fit better in to the existing code?  The code that gets all runtimes or all servers (WebSphereServerInfo) could include the extended ones?  I guess it depends on how expensive the set up is and whether it can be cached.  Some code would have to be changed as well.  Such as in ConfigUtils.getUserDirectory it determines if the file belongs to the user directory when really it should ask the user directory if the file belongs to it.
- Can the names be made more generic such as LibertyRuntimeProvider or maybe even LibertyComponentProvider?

**Comment by keithchong** [comment]
@eharris369, one more thing, I did not implement the extension point and related changes for the Maven plugin.  Wanted to get your ideas on this first before continuing.

eg. In the gradle plugin:

    <extension point="com.ibm.ws.st.core.ghostRuntimeProvider">
         <provider class="com.ibm.ws.st.liberty.gradle.manager.internal.GhostRuntimeProvider"></provider>
    </extension>

**Comment by keithchong** [comment]
This initial PR has been replaced by #168.

Cancelling/Closing.

**Pull request closed without merging**